### PR TITLE
chore: Add `combobox-cancel` example

### PR DESCRIPTION
**Pull request by mitjapotocin**
image

### CODE REVIEW

File: packages/ariakit/src/combobox/__examples__/combobox-cancel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      <label className="label">
+        Your favorite food
+        <div className="combobox-wrapper">
+          <Combobox
+            state={combobox}
+            placeholder="e.g., Apple"
+            className="combobox"
+          />
+          <ComboboxCancel
+            role="button"
```
Comment (diegohaz) [inline packages/ariakit/src/combobox/__examples__/combobox-cancel/index.tsx]: We don't need this prop.

File: packages/ariakit/src/combobox/__examples__/combobox-cancel/test.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { click, getByRole, render } from "ariakit-test-utils";
+import { axe } from "jest-axe";
+import Example from ".";
+
+const getCombobox = () => getByRole("combobox");
```
Comment (diegohaz) [inline packages/ariakit/src/combobox/__examples__/combobox-cancel/test.tsx:5]: Some helpers:
```suggestion
const getCombobox = () => getByRole("combobox");
const getCancelButton = () => getByRole("button", { name: "Clear input" });
const getOption = (name: string) => getByRole("option", { name });
```

File: packages/ariakit/src/combobox/__examples__/combobox-cancel/test.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+test("clear result", async () => {
+  render(<Example />);
+  await click(getCombobox());
+  expect(getCombobox()).toHaveValue("");
+  await click(getByRole("option", { name: "Apple" }));
+  expect(getCombobox()).toHaveFocus();
+  expect(getCombobox()).toHaveValue("Apple");
+  await click(getByRole("button"));
+  expect(getCombobox()).toHaveValue("");
```
Comment (diegohaz) [inline packages/ariakit/src/combobox/__examples__/combobox-cancel/test.tsx]: We have to check if the focus is on the input after clicking on the cancel button.

**Review by diegohaz — COMMENTED** [review]
Hey @mitjapotocin! Thanks for the PR. Here are my suggestions:

- Let's use the simple Combobox example as a base. I mean, we don't need to use `matches` unless we want to test some interaction between matches and the cancel button. There's already a `combobox-matches` example, so people can use it as a reference if they want to combine matches and cancel.

- The cancel button style looks good, but it could be improved. I'd `@import` the button example styles and add `button secondary` class names to the cancel button. `h-8 w-8 p-0 top-1 right-1 rounded` should do the job for the specific `.combobox-cancel` styles.

- I'd rename the test to something like `clear input with mouse` and add a `clear input with keyboard` test (using `press.Tab` and `press.Enter`/`press.Space`).

**Comment by mitjapotocin** [comment]
Hey @diegohaz

Thank you for the review

File: packages/ariakit/src/combobox/__examples__/combobox-cancel/style.css [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+.combobox-wrapper {
+  @apply relative;
+}
+
+.combobox {
+  @apply pr-10;
+}
+
+.combobox-cancel {
+  @apply absolute h-8 w-8 p-0 top-1 right-1;
```
Comment (diegohaz) [inline packages/ariakit/src/combobox/__examples__/combobox-cancel/style.css]:
```suggestion
  @apply absolute h-8 w-8 p-0 top-1 right-1 rounded;
```

File: packages/ariakit/src/combobox/__examples__/combobox-cancel/test.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  expect(getCombobox()).toHaveFocus();
+};
+const selectAnOption = async () => {
+  await click(getCombobox());
+  expect(getCombobox()).toHaveValue("");
+  await click(getOption("Apple"));
+  expect(getCombobox()).toHaveFocus();
+  expect(getCombobox()).toHaveValue("Apple");
+};
```
Comment (diegohaz) [inline packages/ariakit/src/combobox/__examples__/combobox-cancel/test.tsx]: I'd recommend against these abstractions in this case. It's okay to repeat them on each test. The problem with abstracting this way is that failures will generate ambiguous logs. The first lines will not point to the exact tests.

File: packages/ariakit/src/combobox/__examples__/combobox-cancel/list.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export default [
+  "Apple",
+  "Bacon",
+  "Banana",
+  "Broccoli",
+  "Burger",
+  "Cake",
+  "Candy",
+];
```
Comment (diegohaz) [inline packages/ariakit/src/combobox/__examples__/combobox-cancel/list.ts]: We can delete this file and move the list to the index file.

Comment (mitjapotocin) [reply packages/ariakit/src/combobox/__examples__/combobox-cancel/list.ts]: Yes I was thinking about that, no need to have such a small list in a file

**Review by diegohaz — APPROVED** [review]
Thank you! 🚀

**Pull request merged**